docker/package/tezos_setup_wizard.py

```diff
@@ -293,6 +338,7 @@ def get_snapshot_link(self):
             )
 
             self.config["snapshot_url"] = snapshot_metadata["url"]
+            self.config["snapshot_sha256"] = snapshot_metadata.get("sha256", None)
```

Comment:
> If there is no better option than using the file SHA256

I don't see any better option. As you already mentioned, `url` is not reliable source of truth, since its contents could be changed. 

We indeed can't understand if the file we download is the same as the file we intend to download until we fully download it. So that we can only rely on [metadata we can trust](xtz-shots.io/tezos-snapshots.json).

It's possible to ask `sha256` from user, but I believe then we should also check snapshot for integrity.

Reply:
It is exactly what was done in recent commits